### PR TITLE
RESTWS-826: Add REST resource for Address Template Management

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
## Description of what I changed
Added REST Controller for managing Address Template xml.

## Issue I worked on
see https://issues.openmrs.org/browse/RESTWS-826

## Checklist: I completed these to help reviewers :)
- [X] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/AddressTemplateController2_0.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/addresstemplate")
+public class AddressTemplateController2_0 extends BaseRestController {
+
+	@RequestMapping(method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	@ResponseBody()
+	public Object getCurrentConfiguration() {
+		Context.requirePrivilege(PrivilegeConstants.MANAGE_ADDRESS_TEMPLATES);
```
Comment (HerbertYiga) [inline omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/AddressTemplateController2_0.java:45]: thanks @BartlomiejRasztabiga  looking through this

**Comment by dkayiwa** [comment]
@BartlomiejRasztabiga isn't this all that you need? `https://qa-refapp.openmrs.org/openmrs/ws/rest/v1/systemsetting/layout.address.format`

**Comment by BartlomiejRasztabiga** [comment]
> @BartlomiejRasztabiga isn't this all that you need? `https://qa-refapp.openmrs.org/openmrs/ws/rest/v1/systemsetting/layout.address.format`

Hmm, I didn't see that :) However, does POSTing to `systemsetting `perform validation on provided XML? Because my controller does validate incoming XML, so maybe we could leave it.

**Comment by dkayiwa** [comment]
Having all this extra code to maintain, for just validation of a not so important item like address template? What do you think? 😊

**Comment by BartlomiejRasztabiga** [comment]
Yeah, I guess you're right, less code to maintain wins :) I will then close this PR and update the documentation to point to a different URL.

**Comment by dkayiwa** [comment]
@BartlomiejRasztabiga this is all my fault. I should have warned you right from the JIRA ticket, before you embarked on the pull request. So please accept my apology!

**Comment by BartlomiejRasztabiga** [comment]
@dkayiwa No problem, that wasn't a lot of work anyway 😄

**Comment by dkayiwa** [comment]
> that wasn't a lot of work anyway

For a geek like you. 😊

**Comment by BartlomiejRasztabiga** [comment]
Closing because such funcionality already exists under `/openmrs/ws/rest/v1/systemsetting/layout.address.format` endpoint.

**Pull request closed without merging**